packages/components/nodes/chatmodels/ChatNemoGuardrails/ChatNemoGuardrails.ts

```diff
@@ -0,0 +1,119 @@
+import { BaseChatModel, type BaseChatModelParams } from '@langchain/core/language_models/chat_models'
+import { AIMessageChunk, BaseMessage } from '@langchain/core/messages'
+import { BaseChatModelCallOptions } from '@langchain/core/language_models/chat_models'
+import { NemoClient } from './NemoClient'
+import { CallbackManager, CallbackManagerForLLMRun } from '@langchain/core/callbacks/manager'
+import { ChatResult } from '@langchain/core/outputs'
+import { FailedAttemptHandler } from '@langchain/core/utils/async_caller'
+import { getBaseClasses, ICommonObject, INode, INodeData, INodeParams } from '../../../src'
+
+export interface ChatNemoGuardrailsCallOptions extends BaseChatModelCallOptions {
+    /**
+     * An array of strings to stop on.
+     */
+    stop?: string[]
+}
+
+export interface ChatNemoGuardrailsInput extends BaseChatModelParams {
+    configurationId?: string
+    /**
+     * The host URL of the Nemo server.
+     * @default "http://localhost:8000"
+     */
+    baseUrl?: string
+}
+
+class ChatNemoGuardrailsModel extends BaseChatModel<ChatNemoGuardrailsCallOptions, AIMessageChunk> implements ChatNemoGuardrailsInput {
+    configurationId: string
+    id: string
+    baseUrl: string
+    callbackManager?: CallbackManager | undefined
+    maxConcurrency?: number | undefined
+    maxRetries?: number | undefined
+    onFailedAttempt?: FailedAttemptHandler | undefined
+    client: NemoClient
+
+    _llmType(): string {
+        // Add your implementation here
+        return 'nemo-guardrails'
+    }
+
+    _generate(messages: BaseMessage[], options: this['ParsedCallOptions'], runManager?: CallbackManagerForLLMRun): Promise<ChatResult> {
+        const generate = async (messages: BaseMessage[], client: NemoClient): Promise<ChatResult> => {
+            const chatMessages = await client.chat(messages)
+            const generations = chatMessages.map((message) => {
```

Comment:
can we add `await runManager?.handleLLMNewToken(token ?? '')` so that we can trace it for analytic as well

Reply:
Hi @HenryHengZJ

I perceived this await runManager... is used when request is a streming. In this case, my request is direct. Do you think use this command is really necessary?

Reply:
this is used for both, you can do this:
```
await _runManager?.handleLLMNewToken(generationResult.generations?.length ? generationResult.generations[0].text : '')
```

Reply:
I got it. I added this line of code (with some adjusments) and it is working fine. Thanks for this!